### PR TITLE
[MNOE-573] Invoice management

### PR DESCRIPTION
**Pull request by hedudelgado**

### CODE REVIEW

File: src/app/views/invoices/invoice.controller.coffee [hunk 1]
```diff
@@ -1,17 +1,86 @@
...
+  # -----------------------------------------------------------------
+  vm.downloadInvoice = (slug) ->
+    $window.location.href = '/mnoe/admin/invoices/' + slug
 
   # The expected date is the 2nd of the month following the invoice period
   vm.expectedPaymentDate = (endOfInvoiceDate) ->
     DatesHelper.expectedPaymentDate(endOfInvoiceDate)
 
+  vm.changeInvoiceStatus = () ->
+    vm.invoice.paid_at = moment().toISOString()
```
Comment (alexnoox) [inline src/app/views/invoices/invoice.controller.coffee]: `amMoment` service should be used

File: src/app/views/invoices/invoice.html [hunk 2]
```diff
@@ -3,41 +3,144 @@ <h1 translate>mnoe_admin_panel.dashboard.invoice.invoice_info</h1>
...
-          <th class="col-lg-2" translate>mnoe_admin_panel.dashboard.invoice.customer</th>
-          <th class="col-lg-3" translate>mnoe_admin_panel.dashboard.invoice.period</th>
-          <th class="col-lg-2" ng-if="vm.isPaymentEnabled" translate>mnoe_admin_panel.dashboard.invoice.expected_payment_date</th>
-          <th class="col-lg-1" translate>mnoe_admin_panel.dashboard.invoice.amount</th>
-          <th class="col-lg-2" translate>mnoe_admin_panel.dashboard.invoice.status</th>
-          <th class="col-lg-1" translate></th>
+          <th class="col-lg-2" translate>mnoe_admin_panel.dashboard.invoice.details.product</th>
+          <th class="col-lg-2" translate>mnoe_admin_panel.dashboard.invoice.details.type</th>
+          <th class="col-lg-3" translate>mnoe_admin_panel.dashboard.invoice.details.usage</th>
+          <th class="col-lg-1" translate>mnoe_admin_panel.dashboard.invoice.details.price</th>
```
Comment (alexnoox) [inline src/app/views/invoices/invoice.html]: Thre should be 12 cols (8 used here)

File: src/app/views/invoices/invoice.html [hunk 3]
```diff
@@ -3,41 +3,144 @@ <h1 translate>mnoe_admin_panel.dashboard.invoice.invoice_info</h1>
...
-          <td ng-if="vm.isPaymentEnabled">{{vm.expectedPaymentDate(vm.invoice.ended_at) | amCalendar }}</td>
-          <td>{{vm.invoice.price.currency.symbol}}{{vm.invoice.price.fractional}}</td>
-          <td>
-            <span ng-show="vm.invoice.paid_at" translate>mnoe_admin_panel.dashboard.invoices.paid</span>
-            <span ng-show="!vm.invoice.paid_at" translate>mnoe_admin_panel.dashboard.invoices.awaiting</span>
-          </td>
-          <td>
-            <a ng-href="/mnoe/admin/invoices/{{vm.invoice.slug}}" target="_blank"><i class="fa fa-cloud-download"></i></a>
-          </td>
+      <tbody ng-show="!vm.isLoading">
```
Comment (alexnoox) [inline src/app/views/invoices/invoice.html]: The loading is managed by the widget, I think this condition can be removed.

File: src/app/views/invoices/invoice.html [hunk 4]
```diff
@@ -3,41 +3,144 @@ <h1 translate>mnoe_admin_panel.dashboard.invoice.invoice_info</h1>
...
+        <tr ng-repeat="bill in vm.invoice.bills">
+          <td>{{bill.billable_name.attributes.name}}</td>
+          <td>{{bill.billable_type}}</td>
+          <td>{{bill.billing_type}}</td>
+          <td>{{bill.currency}}{{bill.price_cents}}</td>
```
Comment (alexnoox) [inline src/app/views/invoices/invoice.html]: The price is in cents

File: src/app/views/invoices/invoice.html [hunk 5]
```diff
@@ -3,41 +3,144 @@ <h1 translate>mnoe_admin_panel.dashboard.invoice.invoice_info</h1>
...
+                  <i class="fa fa-trash-o"></i>
+                </button>
+              </div>
+            </td>
+          </tr>
+        </tbody>
+      </table>
+    </mno-widget-body>
+    <mno-widget-footer>
+    </mno-widget-footer>
```
Comment (alexnoox) [inline src/app/views/invoices/invoice.html]: The footer is optional, it can be removed if empty.

File: src/app/views/invoices/invoice.html [hunk 6]
```diff
@@ -3,41 +3,144 @@ <h1 translate>mnoe_admin_panel.dashboard.invoice.invoice_info</h1>
...
+  </mno-widget>
+  <div class="row button-footer">
+    <div class="col-xs-12">
+      <div class='btn-group pull-right'>
+        <div class="top-buffer-1" ng-if="!vm.isInvoiceEditedPaid">
+          <button role="button" ui-sref="dashboard.invoices" class="btn btn-primary" translate>Cancel</button>
+          <button role="button" ng-click="vm.update()" class="btn btn-primary" translate>Update</button>
+          <button role="button" ng-click="" class="btn btn-primary" translate>mnoe_admin_panel.dashboard.invoice.adjustments.send_to_customer</button>
+        </div>
```
Comment (alexnoox) [inline src/app/views/invoices/invoice.html]: If needed, this buttons can be moved in the footer.

**Comment by ouranos** [comment]
Closing in favor of #82

**Pull request closed without merging**